templates/cli/lib/questions.js.twig

```diff
@@ -768,7 +772,66 @@ const questionsMfaChallenge = [
     }
 ];
 
+const questionsRunFunctions = [
+    {
+        type: "list",
+        name: "function",
+        message: "Which function would you like to develop locally?",
+        validate: (value) => validateRequired('function', value),
+        choices: () => {
+            let functions = localConfig.getFunctions();
+            if (functions.length === 0) {
+                throw new Error("No functions found in the current directory.");
+            }
+            let choices = functions.map((func, idx) => {
+                return {
+                    name: `${func.name} (${func.$id})`,
+                    value: func.$id
+                }
+            })
+            return choices;
+        }
+    },
+    {
+        type: "input",
+        name: "port",
+        message: 'Which port would you like function to listen on?',
+        default: async () => {
+            let port = 3000;
+            while(port < 3100) {
+                const taken = await isPortTaken(port);
+                if(!taken) {
+                    return port;
+                }
+
+                port++;
+            }
+
+            return 3000;
+        },
+        validate: function(value) {
+            const done = this.async();
+
+            (async () => {
+                if (typeof value !== 'number' && isNaN(+value)) {
+                    throw Error(`You need to provide a number.`);
```

Comment:
```suggestion
                    throw Error(`Port needs to be a valid integer between 1 and 65536`);
```
Even though ports 1 to 65536 are available in theory, some ranges are reserved for specific use cases. So lets research what is the allowed port range and modify the error message and condition to reflect it

Reply:
I think 1024 and upwards should be allow for custom use, below are considered system ports